crates/node/src/lib.rs

```diff
@@ -986,6 +1032,112 @@ impl Node {
+                types::CatchupStreamMessage::Response(response) => {
+                    let Some(ref context) = context else {
+                        eyre::bail!("Received response with transactions before meta")
+                    };
+
+                    for transaction in response.transactions {
+                        match transaction.status {
+                            types::TransactionStatus::Pending => {
+                                self.tx_pool.insert(
+                                    chosen_peer,
+                                    calimero_primitives::transaction::Transaction {
+                                        context_id: context.id,
+                                        method: transaction.transaction.method,
+                                        payload: transaction.transaction.payload,
+                                        prior_hash: transaction.transaction.prior_hash,
+                                    },
+                                    None,
+                                )?;
+                            }
+                            types::TransactionStatus::Executed => {
+                                self.execute_transaction(
+                                    context.clone(),
+                                    transaction.transaction,
+                                    transaction.transaction_hash,
+                                )
+                                .await?;
+                            }
+                        }
+                    }
+                }
+                types::CatchupStreamMessage::ResponseMeta(response) => match context {
+                    Some(ref context) => {
+                        if context.application_id != response.application_id {
+                            eyre::bail!(
+                                "Application ID mismatch: expected {:?}, got {:?}",
+                                context.application_id,
+                                response.application_id
+                            );
+                        }
+                    }
+                    None => {
+                        self.ctx_manager
+                            .install_application(&response.application_id, &response.version)
+                            .await?;
+
+                        let context_ = calimero_primitives::context::Context {
+                            id: context_id,
+                            application_id: response.application_id,
+                        };
+
+                        self.ctx_manager.add_context(context_.clone()).await?;
+                        context = Some(context_);
+                    }
+                },
+                types::CatchupStreamMessage::Error(err) => {
+                    eyre::bail!(err);
+                }
+                event => {
+                    warn!(?event, "Unexpected event");
+                }
+            }
+        }
+
+        Ok(Some(()))
```

Comment:
can't find any `return Ok(None)` so this can be simplified to return `eyre::Result<()>`